frontend/src/scenes/surveys/SurveyCustomization.tsx

```diff
@@ -130,7 +131,34 @@ export function Customization({ appearance, surveyQuestionItem, onAppearanceChan
                                 <span>Shuffle questions</span>
                             </div>
                         }
-                        onChange={(checked) => onAppearanceChange({ ...appearance, shuffleQuestions: checked })}
+                        onChange={(checked) => {
+                            if (checked && hasBranchingLogic) {
+                                onAppearanceChange({ ...appearance, shuffleQuestions: false })
```

Comment:
LemonCheckbox behaves like an uncontrolled component when the dialog is opened, so this needs to be set to false to make sure it stays unchecked initially.